cloudant-sync-datastore-android/src/test/java/com/cloudant/sync/replication/WifiPeriodicReplicationReceiverTest.java

```diff
@@ -0,0 +1,210 @@
+    /**
+     * Check that when {@link WifiPeriodicReplicationReceiver} receives
+     * {@link ConnectivityManager#CONNECTIVITY_ACTION} and WiFi is connected,
+     * an {@link Intent} is sent out to start the Service
+     * {@link ReplicationService} associated with
+     * {@link WifiPeriodicReplicationReceiver} containing the extra
+     * {@link ReplicationService#EXTRA_COMMAND} with the value
+     * {@link PeriodicReplicationService#COMMAND_START_PERIODIC_REPLICATION}.
+     */
+    public void testWifiConnected() {
+        Intent intent = new Intent(ConnectivityManager.CONNECTIVITY_ACTION);
+
+        mMockContext.setMockConnectivityManager(ConnectivityManager.TYPE_WIFI, true);
+
+        mReceiver.onReceive(mMockContext, intent);
+        assertEquals(1, mMockContext.getIntentsReceived().size());
+
+        Intent receivedIntent = mMockContext.getIntentsReceived().get(0);
+        assertEquals(ReplicationService.class.getName(), receivedIntent.getComponent().getClassName());
+        assertNull(receivedIntent.getAction());
+        assertEquals(PeriodicReplicationService.COMMAND_START_PERIODIC_REPLICATION, receivedIntent.getIntExtra(ReplicationService.EXTRA_COMMAND, ReplicationService.COMMAND_NONE));
+    }
+
+    /**
+     * Check that when {@link WifiPeriodicReplicationReceiver} receives
+     * {@link ConnectivityManager#CONNECTIVITY_ACTION} and WiFi is not connected,
+     * an {@link Intent} is sent out to start the Service
+     * {@link ReplicationService} associated with
+     * {@link WifiPeriodicReplicationReceiver} containing the extra
+     * {@link ReplicationService#EXTRA_COMMAND} with the value
+     * {@link PeriodicReplicationService#COMMAND_START_PERIODIC_REPLICATION}.
+     */
+    public void testWifiDisconnected() {
+        Intent intent = new Intent(ConnectivityManager.CONNECTIVITY_ACTION);
+
+        mMockContext.setMockConnectivityManager(ConnectivityManager.TYPE_WIFI, false);
+
+        mReceiver.onReceive(mMockContext, intent);
+        assertEquals(1, mMockContext.getIntentsReceived().size());
+
+        Intent receivedIntent = mMockContext.getIntentsReceived().get(0);
+        assertEquals(ReplicationService.class.getName(), receivedIntent.getComponent().getClassName());
+        assertNull(receivedIntent.getAction());
+        assertEquals(PeriodicReplicationService.COMMAND_STOP_PERIODIC_REPLICATION, receivedIntent.getIntExtra(ReplicationService.EXTRA_COMMAND, ReplicationService.COMMAND_NONE));
+    }
+
+    /**
+     * Check that when {@link WifiPeriodicReplicationReceiver} receives
+     * {@link ConnectivityManager#CONNECTIVITY_ACTION} and the device is connected to
+     * a non-WiFi network, an {@link Intent} is sent out to start the Service
+     * {@link ReplicationService} associated with
+     * {@link WifiPeriodicReplicationReceiver} containing the extra
+     * {@link ReplicationService#EXTRA_COMMAND} with the value
+     * {@link PeriodicReplicationService#COMMAND_START_PERIODIC_REPLICATION}.
```

Comment:
This does not match up with the assertion, I'm assuming the assertion is correct given that this is a Wifi replicator.